File: app/src/main/java/com/hencoder/praticedraw6/MainActivity.java
package com.hencoder.praticedraw6;

import android.os.Bundle;
import android.view.Menu;

import androidx.annotation.LayoutRes;
import androidx.annotation.StringRes;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.ViewPager;

import com.google.android.material.tabs.TabLayout;
import com.hencoder.hencoderpratice.R;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    TabLayout tabLayout;
    ViewPager pager;
    List<PageModel> pageModels = new ArrayList<>();

    {
        pageModels.add(new PageModel(R.layout.sample6_translation, R.string.title_translation, R.layout.practice6_translation));
        pageModels.add(new PageModel(R.layout.sample6_rotation, R.string.title_rotation, R.layout.practice6_rotation));
        pageModels.add(new PageModel(R.layout.sample6_scale, R.string.title_scale, R.layout.practice6_scale));
        pageModels.add(new PageModel(R.layout.sample6_alpha, R.string.title_alpha, R.layout.practice6_alpha));
        pageModels.add(new PageModel(R.layout.sample6_multi_properties, R.string.title_multi_properties, R.layout.practice6_multi_properties));
        pageModels.add(new PageModel(R.layout.sample6_duration, R.string.title_duration, R.layout.practice6_duration));
        pageModels.add(new PageModel(R.layout.sample6_interpolator, R.string.title_interpolator, R.layout.practice6_interpolator));
        pageModels.add(new PageModel(R.layout.sample6_object_anomator, R.string.title_object_animator, R.layout.practice6_object_animator));
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        pager = (ViewPager) findViewById(R.id.pager);
        pager.setAdapter(new FragmentPagerAdapter(getSupportFragmentManager()) {

            @Override
            public Fragment getItem(int position) {
                PageModel pageModel = pageModels.get(position);
                return PageFragment.newInstance(pageModel.sampleLayoutRes, pageModel.practiceLayoutRes);
            }

            @Override
            public int getCount() {
                return pageModels.size();
            }

            @Override
            public CharSequence getPageTitle(int position) {
                return getString(pageModels.get(position).titleRes);
            }
        });

        tabLayout = (TabLayout) findViewById(R.id.tabLayout);
        tabLayout.setupWithViewPager(pager);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        return super.onCreateOptionsMenu(menu);
    }

    private class PageModel {
        @LayoutRes
        int sampleLayoutRes;
        @StringRes
        int titleRes;
        @LayoutRes int practiceLayoutRes;

        PageModel(@LayoutRes int sampleLayoutRes, @StringRes int titleRes, @LayoutRes int practiceLayoutRes) {
            this.sampleLayoutRes = sampleLayoutRes;
            this.titleRes = titleRes;
            this.practiceLayoutRes = practiceLayoutRes;
        }
    }
}

File: app/src/main/java/com/hencoder/praticedraw4/MainActivity.kt
package com.hencoder.praticedraw4

import android.os.Bundle
import android.view.Menu
import android.view.View
import androidx.annotation.LayoutRes
import androidx.annotation.StringRes
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentPagerAdapter
import androidx.viewpager.widget.ViewPager
import com.google.android.material.tabs.TabLayout
import com.hencoder.hencoderpratice.R

class MainActivity : AppCompatActivity() {
  var tabLayout: TabLayout? = null
  var pager: ViewPager? = null
  var pageModels: MutableList<PageModel> = ArrayList()

  init {
    pageModels.add(
      PageModel(
        R.layout.sample_clip_rect,
        R.string.title_clip_rect,
        R.layout.practice_clip_rect
      )
    )
    pageModels.add(
      PageModel(
        R.layout.sample_clip_path,
        R.string.title_clip_path,
        R.layout.practice_clip_path
      )
    )
    pageModels.add(
      PageModel(
        R.layout.sample_translate,
        R.string.title_translate,
        R.layout.practice_translate
      )
    )
    pageModels.add(PageModel(R.layout.sample_scale, R.string.title_scale, R.layout.practice_scale))
    pageModels.add(
      PageModel(
        R.layout.sample_rotate,
        R.string.title_rotate,
        R.layout.practice_rotate
      )
    )
    pageModels.add(PageModel(R.layout.sample_skew, R.string.title_skew, R.layout.practice_skew))
    pageModels.add(
      PageModel(
        R.layout.sample_matrix_translate,
        R.string.title_matrix_translate,
        R.layout.practice_matrix_translate
      )
    )
    pageModels.add(
      PageModel(
        R.layout.sample_matrix_scale,
        R.string.title_matrix_scale,
        R.layout.practice_matrix_scale
      )
    )
    pageModels.add(
      PageModel(
        R.layout.sample_matrix_rotate,
        R.string.title_matrix_rotate,
        R.layout.practice_matrix_rotate
      )
    )
    pageModels.add(
      PageModel(
        R.layout.sample_matrix_skew,
        R.string.title_matrix_skew,
        R.layout.practice_matrix_skew
      )
    )
    pageModels.add(
      PageModel(
        R.layout.sample_camera_rotate,
        R.string.title_camera_rotate,
        R.layout.practice_camera_rotate
      )
    )
    pageModels.add(
      PageModel(
        R.layout.sample_camera_rotate_fixed,
        R.string.title_camera_rotate_fixed,
        R.layout.practice_measure_text
      )
    )
    pageModels.add(
      PageModel(
        R.layout.sample_camera_rotate_hitting_face,
        R.string.title_camera_rotate_hitting_face,
        R.layout.practice_camera_rotate_hitting_face
      )
    )
    pageModels.add(
      PageModel(
        R.layout.sample_flipboard,
        R.string.title_flipboard,
        R.layout.practice_flipboard
      )
    )
  }

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)
    pager = findViewById<View>(R.id.pager) as ViewPager
    pager!!.adapter = object : FragmentPagerAdapter(supportFragmentManager) {
      override fun getItem(position: Int): Fragment {
        val pageModel = pageModels[position]
        return PageFragment.newInstance(
          pageModel.sampleLayoutRes,
          pageModel.practiceLayoutRes
        )
      }


      override fun getCount(): Int {
        return pageModels.size
      }

      override fun getPageTitle(position: Int): CharSequence {
        return getString(pageModels[position].titleRes)
      }
    }
    tabLayout = findViewById<View>(R.id.tabLayout) as TabLayout
    tabLayout?.setupWithViewPager(pager)
  }

  override fun onCreateOptionsMenu(menu: Menu): Boolean {
    return super.onCreateOptionsMenu(menu)
  }

  inner class PageModel internal constructor(
    @field:LayoutRes @param:LayoutRes var sampleLayoutRes: Int,
    @field:StringRes @param:StringRes var titleRes: Int,
    @field:LayoutRes @param:LayoutRes var practiceLayoutRes: Int
  )
}